src/System.Drawing.Common/tests/Imaging/EncoderParameterTests.cs

```diff
@@ -0,0 +1,319 @@
+        [InlineData(new long[] { 0, 1 }, null, typeof(NullReferenceException))]
+        [InlineData(null, new long[] { 0, 1 }, typeof(NullReferenceException))]
+        public void Ctor_RangeBegin_RangeEnd_InvalidValues_ThrowsExpected(long[] rangeBegin, long[] rangeEnd, Type expected)
+        {
+            Assert.Throws(expected, () => new EncoderParameter(AnyEncoder, rangeBegin, rangeEnd));
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void Ctor_Encoder_NullString_ThrowsNullReferenceException()
+        {
+            Assert.Throws<NullReferenceException>(() => new EncoderParameter(AnyEncoder, (string)null));
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void Ctor_Encoder_ByteArray_ThrowsNullReferenceException()
+        {
+            Assert.Throws<NullReferenceException>(() => new EncoderParameter(AnyEncoder, (byte[])null));
+        }
+
+        public static IEnumerable<object[]> EncoderParameterCtor_NullEncoder_TestData
+        {
+            get
+            {
+                yield return new object[] { new Action(() => new EncoderParameter(null, (byte)0)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, (byte)0, false)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, (short)0)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, numerator: 0, denominator: 0)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, rangebegin: 0, rangeend: 0)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, 0, 0, 0, 0)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, "anyString")) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new byte[] { })) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new byte[] { }, false)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new short[] { })) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new long[] { })) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new int[] { }, new int[] { })) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new long[] { }, new long[] { })) };
+            }
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [MemberData(nameof(EncoderParameterCtor_NullEncoder_TestData))]
+        public void Ctor_NullEncoder_ThrowsNullReferenceException(Action encoderCtor)
```

Comment:
This seems unecessary - I know in general we/I say "use theory", but I'm not sure it's the best place for it here.

I'd just do 

```cs
public void Ctor_NullEncoder_ThrowsNullReferenceException(Action encoderCtor)
{
    Assert.Throws<NRE>(...):
    Assert.Throws<NRE>(...):
    Assert.Throws<NRE>(...):
    //...
}
``